### ISSUE TITLE
Parser panic

### ISSUE THREAD

**Issue by bufdev** [jhump/protoreflect#339]
New one came up:

https://github.com/bufbuild/buf/issues/107

https://github.com/michilu/buf.build-reports/tree/issue-107

```
$ buf image build -o /dev/null
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1599e47]

goroutine 19 [running]:
github.com/jhump/protoreflect/dynamic.(*Message).findMissingFields(0x0, 0x80, 0x5d, 0x1ad8d20)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2592 +0x37
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive(0x0, 0xc000041980, 0x5f, 0x5f, 0xc00028d0e8)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2615 +0x43
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive.func1(0x1ac2fc0, 0x0, 0x0, 0x1ac2fc0)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2636 +0x63
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive(0xc000698330, 0xc0001e04b0, 0x48, 0x48, 0xc00028d390)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2647 +0x4a1
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive.func1(0x1ac2fc0, 0xc000698330, 0xc000698330, 0x1ac2fc0)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2636 +0x63
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive(0xc000698300, 0xc000272a00, 0x3c, 0x3c, 0xc00028d638)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2657 +0x924
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive.func1(0x1ac2fc0, 0xc000698300, 0xc000698300, 0x1ac2fc0)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2636 +0x63
github.com/jhump/protoreflect/dynamic.(*Message).validateRecursive(0xc000371f80, 0x0, 0x0, 0xc000454288, 0xc000371f80)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2663 +0xadb
github.com/jhump/protoreflect/dynamic.(*Message).ValidateRecursive(...)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/dynamic/dynamic_message.go:2611
github.com/jhump/protoreflect/desc/protoparse.interpretOptions(0xc00048e750, 0x2749998, 0xc0001d9180, 0x1ac3fc0, 0xc0004ac000, 0xc00041e620, 0x1, 0x1, 0x15e1a22, 0xc0004ac960, ...)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/desc/protoparse/options.go:988 +0xab4
github.com/jhump/protoreflect/desc/protoparse.interpretFileOptions(0xc00048e750, 0x1ad2e20, 0xc0001d9180, 0x12, 0xc0000b2580)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/desc/protoparse/options.go:685 +0x6ae
github.com/jhump/protoreflect/desc/protoparse.(*linker).linkFiles(0xc0006a0cc0, 0xc0006a0cc0, 0x3, 0x3)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/desc/protoparse/linker.go:66 +0x1af
github.com/jhump/protoreflect/desc/protoparse.Parser.ParseFiles(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00040e6d0, 0x1, 0xc0004725c0, 0x0, ...)
	/go/pkg/mod/github.com/jhump/protoreflect@v1.7.1-0.20200713144533-3216fce50c70/desc/protoparse/parser.go:225 +0x2cd
github.com/bufbuild/buf/internal/buf/bufbuild.getBuildResult(0x1ac7380, 0xc00042e180, 0xc00044e190, 0xc00048e360, 0x3, 0x3, 0x0, 0x0)
	/Users/pedge/git/bufbuild/internal/buf/bufbuild/builder.go:165 +0x176
github.com/bufbuild/buf/internal/buf/bufbuild.(*builder).getBuildResults.func1(0xc00042ef00, 0x1ac7380, 0xc00042e180, 0xc00044e190, 0xc00048e360, 0x3, 0x3, 0x203000)
	/Users/pedge/git/bufbuild/internal/buf/bufbuild/builder.go:124 +0x73
created by github.com/bufbuild/buf/internal/buf/bufbuild.(*builder).getBuildResults
	/Users/pedge/git/bufbuild/internal/buf/bufbuild/builder.go:123 +0x211
```

**Comment by johanbrandhorst** [comment]
Manage to reproduce this locally with the following:
```
package main

import "github.com/jhump/protoreflect/desc/protoparse"

func main() {
	parser := &protoparse.Parser{
		ImportPaths: []string{
			".",
		},
	}

	parser.ParseFiles("proto/option.proto")
}
```

Based on folder structure from https://github.com/michilu/buf.build-reports/tree/issue-107

**Comment by jhump** [comment]
Thanks for the repro details! I'll dig in this afternoon.

**Comment by jhump** [comment]
Fixed in #340
